### PR TITLE
Multiple backend corrections

### PR DESCRIPTION
**Pull request by yabirgb**
- Fixes double '0x' in location for ws message
- Fixes FK error related to spam asset being omitted in the user's db during addition

### CODE REVIEW

File: rotkehlchen/data_migrations/migrations/migration_8.py [hunk 1]
```diff
@@ -67,8 +67,7 @@ def data_migration_8(rotki: 'Rotkehlchen', progress_handler: 'MigrationProgressH
                 message_type=WSMessageType.EVM_ADDRESS_MIGRATION,
                 data=[{'evm_chain': str(x[0]), 'address': x[1]} for x in to_add_accounts],
             )
-    else:
-        # The only step is update the spam assets
+    else:  # The only step is update the spam assets
```
Comment (LefterisJP) [inline rotkehlchen/data_migrations/migrations/migration_8.py:70]: Pay closer attention to what I say in the PR reviews.

```suggestion
    else:  # The only step is updating the spam assets
```

File: rotkehlchen/assets/spam_assets.py [hunk 2]
```diff
@@ -1068,6 +1068,9 @@ def query_token_spam_list() -> set[Asset]:
                 asset_type=AssetType.EVM_TOKEN,
                 data=evm_token,
             )
+            # add the asset to the asset table in the user's db
+            with db.user_write() as cursor:
```
[the remark this answers is not in the document]
Comment (yabirgb) [reply rotkehlchen/assets/spam_assets.py:1072]: If it already exists it means it was in the globaldb before the unlock and in that case we copy it already from the globaldb https://github.com/rotki/rotki/blob/345c33127bab4cc86342adf8d7efca71e28ce6d4/rotkehlchen/db/dbhandler.py#L2430

[the remark this answers is not in the document]
Comment (LefterisJP) [reply rotkehlchen/assets/spam_assets.py:1072]: all right

**Review by LefterisJP — APPROVED** [review]
LGTM. Thank you @yabirgb

**Pull request merged**